libraries/Microsoft.Bot.Configuration/BotConfigurationExtensions.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    public static class BotConfigurationExtensions
+    {
+        /// <summary>
+        /// Returns a IEnumerable of a specific <see cref="ConnectedService"/> type contained in the bot configuration file.
+        /// </summary>
+        /// <param name="botServices">BotConfiguration's services.</param>
+        /// <param name="serviceType">The type of <see cref="ConnectedService"/> to search.</param>
+        /// <returns>List of bot's <see cref="ConnectedService"/>.</returns>
+        public static IEnumerable<ConnectedService> OfType(this List<ConnectedService> botServices, string serviceType)
+        {
+            // If botServices is null, throws an exception
+            if (botServices == null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            // If there aren't any service, returns an empty IEnumerable
+            if (botServices.Count == 0)
+            {
+                return Enumerable.Empty<ConnectedService>();
+            }
+
+            // Obtains all the services of the specified type
+            var botFilteredServices = botServices.Where(service => service.Type.Equals(serviceType, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
I don't see why you'd want to do the comparison based on the "current" culture. These are programmatic values, not user input, so you can just use `OrdinalIgnoreCase`.